Sources/LiveKit/Core/SignalClient.swift

```diff
@@ -78,20 +85,53 @@ internal class SignalClient: MulticastDelegate<SignalClientDelegate> {
         }
 
         latestJoinResponse = nil
+        queue.removeAll()
     }
 }
 
 // MARK: - Private
 
 private extension SignalClient {
 
-    func sendRequest(_ request: Livekit_SignalRequest) -> Promise<Void> {
+    func sendQueuedRequests() -> Promise<Void> {
+
+        // quickly return if no queued requests
+        guard !queue.isEmpty else {
+            log("No queued requests")
+            return Promise(())
+        }
+
+        // create a promise that never throws so the send sequence can continue
+        func safeSend(_ request: Livekit_SignalRequest) -> Promise<Void> {
+            sendRequest(request, enqueueIfReconnecting: false).recover(on: .sdk) { error in
+                self.log("Failed to send queued request, request: \(request) \(error)", .warning)
+            }
+        }
+
+        // send requests in sequential order
+        let promises = queue.reduce(into: Promise(())) { result, request in result.then(on: .sdk) { safeSend(request) } }
+        // clear the queue
+        queue.removeAll()
+
+        return promises
+    }
+
+    // send request or enqueue while reconnecting
+    func sendRequest(_ request: Livekit_SignalRequest, enqueueIfReconnecting: Bool = true) -> Promise<Void> {
+
+        guard !(connectionState.isReconnecting && enqueueIfReconnecting) else {
```

Comment:
I think this also has to enqueue when the queue is not yet empty. This is to ensure that we don't send requests out of order.